### ISSUE TITLE
Guide for simple fix of "Tradingview UI Changed" Performance headers error

### ISSUE THREAD

**Issue by stefantabakovq** [akumidv/tradingview-assistant-chrome-extension#58]
This fix works most times, but not all, UI changes can have different aspects! 
This guide is for fixing path of HEADERS / ROWS issue when backtesting!

Apply same logic for different issues, such as UI changes for backtesting tab, strategy settings tab etc.
Basically, check selector.js file, follow previous path for given tab / button and update as follows.

**I recommend turning off WiFi while doing it, as tab auto-refreshes and it's hard to follow.**

1. Press **Strategy Tester > Performance Summary** tab on TV. (Again, this fix is for perf headers issue)
2. Press F12 on Chrome to open up Inspect Element page, navigate to: 
![bottom_area](https://user-images.githubusercontent.com/65942717/185444254-09a9f7c5-8303-41eb-bf21-a3315872759c.png)
3. Navigate to performance table, look for `<thead>` element, this is performance table Headers. Right-click and copy selector of `<th>` element as follows:
![headers](https://user-images.githubusercontent.com/65942717/185444335-1eb1b10d-b7e9-484a-b039-8132b0e96c66.png)
4. Same process for `<tbody>`, however, copy selector of `<tr>`, not any of the `<tb>` elements! 
![row](https://user-images.githubusercontent.com/65942717/185445012-f17f60eb-706d-4315-8fc9-365567ba869f.png)
5. Paste new selectors respectfully into selector.js file, for header it should look like - `#bottom-area > div.bottom-widgetbar-content.backtesting > div > div.backtesting-content-wrapper.widgetContainer-Ywsi3NS8 > div > div > div > div > table > thead > tr > th` without any nth-child extension.
and for body - `#bottom-area > div.bottom-widgetbar-content.backtesting > div > div.backtesting-content-wrapper.widgetContainer-Ywsi3NS8 > div > div > div > div > table > tbody > tr`. Should be similar, not the same always! 
![selector](https://user-images.githubusercontent.com/65942717/185445081-049b5971-a84c-4629-8ce5-1ab32b346681.png)
6. Install manually from chrome://extentions & refresh page.

If error message is NOT about headers / body of performance table, try to get new selectors of given updated element in the same way! 

Other issues need more precise updating, who knows what TV will do!

**Comment by akumidv** [comment]
I fixed an error with performance report in 1.15 version and sent to review it today.
All base function of extension working today on version from repository. 

PS It is not good idea to set full path selectors with structure like `#bottom-area > div.bottom-widgetbar-content.backtesting > div > div.backtesting-content-wrapper.widgetContainer-Ywsi3NS8 > div > div > div > div > table > thead > tr > th` because the element structure and full class list for elements probably change more early than key classes names or data tags.

**Comment by stefantabakovq** [comment]
> I fixed an error with performance report in 1.15 version and sent to review it today. All base function of extension working today on version from repository.
> 
> PS It is not good idea to set full path selectors with structure like `#bottom-area > div.bottom-widgetbar-content.backtesting > div > div.backtesting-content-wrapper.widgetContainer-Ywsi3NS8 > div > div > div > div > table > thead > tr > th` because the element structure and full class list for elements probably change more early than key classes names or data tags.

I agree, I was just showing a quick fix for the time being because it was requested in #57 . 
Thank you for making the code open-source and for your work!

**Comment by duocyber** [comment]
Is it possible to implement an UI selectors generator function inside this extension for those backtesting params by analysing the UI from HTML so that we can quickly use the generated files to patch it on demand?

**Comment by MrDenfish** [comment]
Works great with a premium account...Thanks everyone for your work.

**Comment by xlajfup** [comment]
![未命名](https://user-images.githubusercontent.com/107462217/185760453-e65c6e61-7024-458d-9d7b-188a3b752268.png)
Hai brother
Please how to solve this problem

**Comment by Quantbrinx** [comment]
any idea when we get an update straight from the extension? Messing with the code is a pain. I rely on this app on a daily basis. Getting an update asap would be much great and much appreciated!

**Comment by stefantabakovq** [comment]
> any idea when we get an update straight from the extension? Messing with the code is a pain. I rely on this app on a daily basis. Getting an update asap would be much great and much appreciated!

The developer has made this extension free and open source, he isn't working for us, we aren't paying him, either learn to update if you need it urgently or be patient my man. Also, Google reviews updates slowly and if you rely on it, learn to use it and modify it.

**Comment by akumidv** [comment]
> Hi bro
> Please how to solve this problem

Hi, this is new error and would be more helpful create new issue. I can't understand reason from your screenshot. Can you press F12 and open console tab and after that repeat action that invoke this error? And will made screenshot with tab? Also would helpful if you extend line with error.

**Comment by akumidv** [comment]
> Is it possible to implement an UI selectors generator function inside this extension for those backtesting params by analysing the UI from HTML so that we can quickly use the generated files to patch it on demand?

You can install extension from repository and fix it as you want. Or fork it and fixing your version. All possibilities in your hand – this is an open source app.

**Comment by akumidv** [comment]
> > any idea when we get an update straight from the extension? Messing with the code is a pain. I rely on this app on a daily basis. Getting an update asap would be much great and much appreciated!
> 
> The developer has made this extension free and open source, he isn't working for us, we aren't paying him, either learn to update if you need it urgently or be patient amigo. Also, Google reviews updates slowly and if you rely on it, learn to use it and modify it.

Thanks! I was in vacation on August. Now I can fix problem more faster.
